denizen_scripts/survival/repo-link/items/system/soul_item_system.dsc

```diff
@@ -308,86 +418,115 @@ item_system_build_item:
   definitions: item
   script:
-    - define level <[item].nbt[soul_level]||0>
-    - if <[level]> != 0:
+  # % ██ [ Determine the amount of Stars  ] ██
+    - if <[item].has_nbt[soul_level]>:
+      - define level <[item].nbt[soul_level]>
       - define level_stars <list[].pad_right[<[level]>].with[<&e>✭].pad_right[5].with[<&7>✭].unseparated>
     - else:
+      - define level 0
       - define level_stars <list[].pad_right[5].with[<&7>✭].unseparated>
-    - define base <script[item_system_global_data].yaml_key[defaults.damage.<[item].material.name>]||0>
-    - if <[item].has_nbt[buffs]> && <[item].material.name> != <script[soul].yaml_key[material]> && <[item].nbt[buffs]||none> != none:
-      - foreach <[item].nbt[buffs].unescaped.as_list> as:buff:
-        - if <script[item_system_global_data].yaml_key[nbt_attributes.<[buff].before[,]>]||null> != null:
-          - define attribute <script[item_system_global_data].yaml_key[nbt_attributes.<[buff].before[,]>]>
-          - define slot <script[item_system_global_data].yaml_key[nbt_slots.<[item].material.name>]>
-          - if <[buff].before[,]> == melee_damage:
-            - define value <[buff].after[,].+[<[base]>]>
-          - else:
-            - define value <[buff].after[,]>
-          - define nbt_attributes:|:<[attribute]>/<[slot]>/0/<[value]>
-        - else:
-          - define value <[buff].after[,]>
-          - define nbt:|:<script[item_system_global_data].yaml_key[nbt_other.<[buff].before[,]>]>/<[value]>
-    - if <[item].has_nbt[debuffs]> && <[item].material.name> != <script[soul].yaml_key[material]> && <[item].nbt[debuffs]||none> != none:
-      - foreach <[item].nbt[debuffs].unescaped.as_list> as:debuff:
-        - if <script[item_system_global_data].yaml_key[nbt_attributes.<[debuff].before[,]>]||null> != null:
-          - define attribute <script[item_system_global_data].yaml_key[nbt_attributes.<[debuff].before[,]>]>
-          - define slot <script[item_system_global_data].yaml_key[nbt_slots.<[item].material.name>]>
-          - if <[debuff].before[,]> == melee_damage:
-            - define value <[debuff].after[,].+[<[base]>]>
+
+  # % ██ [ Determine the base of the item?  ] ██
+    - if <script[item_system_global_data].list_keys[defaults.damage].contains[<[item].material.name>]>:
+      - define base <script[item_system_global_data].data_key[defaults.damage.<[item].material.name>]>
+    - else:
+      - define base 0
+
+
+  # % ██ [ NOTES FOR CONSTRUCTING THE MAP FROM THE LIST OF MAPS  ] ██
+      #| 1) foreach the list, foreach the map
+      #^  - foreach <[item].nbt[buffs]> as:buffs:
+      #^    - foreach <[buff]> key:buff as:stat:
+      #| 2) foreach the list, construct a fuller map
+      #^  - define map <map>
+      #^  - foreach <[item].nbt[buffs]> as:buffs:
+      #^    - define map <[map].include[<[buffs]>]>
+      #| 3) construct the map from the list
+      #^  - foreach <[item].nbt[<[modifier]>].parse[to_list].combine.to_map> key:alt as:stat:
```

Comment:
these are notes explicitly for determining how we should be looping across the list of buffs and debuffs; Ideally, a map for each modifier containing the combined map would be the best option, but option `#2` was not possible because `MapTag.include[<Map>]` does not support a list of maps. Option `#3` deconstructs the maps to lists, combines the lists within the list, and reconstructs a map, but is two steps heavier than what option `#2` would benefit.